src/librustc_mir/build/expr/into.rs

```diff
@@ -157,11 +157,8 @@ impl<'a,'tcx> Builder<'a,'tcx> {
                     }
 
                     // execute the body, branching back to the test
```

Comment:
and that we think this is ok because (a) the type of `destination` should always be `()` and (b) the destination will still be considered uninitialized for the first iteration, so it should not cause anything to type-check that wouldn't otherwise